gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/operations/SortedMap2IndexInsertOperation.java

```diff
@@ -284,6 +285,29 @@ else if (lockTimeout > 0) { // wait before throwing conflict
           }
           break;
         }
+
+        // TODO: identify the cause of Bug SNAP-2627
+        // Till then this is the crude fix
+        if (oldValue instanceof AbstractRegionEntry &&
+            ((AbstractRegionEntry)oldValue).isDestroyedOrRemoved()) {
+          // create a dummy exception & throw to get stack:
+          try {
+            throw GemFireXDUtils.newDuplicateKeyViolation("unique constraint",
```

Comment:
I think we can just do this instead of throwing exception:
Exception ex = GemFireXDUtils.newDuplicateKeyViolation("unique constraint",
container.getQualifiedTableName(), "key=" + key.toString()
                     + ", row=" + value, oldValue, null, null);
logger.error("Unique index constraint violation caused due to " +
"removed/destroyed entry. The index is corrupted. Cleaning the index", ex);

----------------------

Any reason why throw and catch and then use? The above should be fine, no?